python/dbsp/config.py

```diff
@@ -17,92 +18,169 @@
 from dbsp_api_client.models.csv_parser_config import CsvParserConfig
 from dbsp_api_client.models.new_config_request import NewConfigRequest
 from dbsp_api_client.models.update_config_request import UpdateConfigRequest
+from dbsp_api_client.models.new_connector_request import NewConnectorRequest
+from dbsp_api_client.models.new_connector_response import NewConnectorResponse
+from dbsp_api_client.models.attached_connector import AttachedConnector
+from dbsp_api_client.models.direction import Direction
+from dbsp_api_client.models.connector_type import ConnectorType
 from dbsp_api_client.models.csv_encoder_config import CsvEncoderConfig
 from dbsp_api_client.api.config import new_config
 from dbsp_api_client.api.config import update_config
 from dbsp_api_client.api.pipeline import new_pipeline
+from dbsp_api_client.api.connector import new_connector
 from dbsp.pipeline import DBSPPipeline
 from dbsp.project import DBSPProject
 from dbsp.error import TimeoutException
+from dbsp.connector import DBSPConnector
+
 
 class DBSPPipelineConfig:
     """Pipeline configuration specified by the user when creating
     a new pipeline instance."""
 
-    def __init__(self, project: DBSPProject, workers: int):
+    def __init__(self, project: DBSPProject, workers: int, name: str = '<anon>', description: str = ''):
         self.project = project
         self.api_client = self.project.api_client
         self.pipeline_config = PipelineConfig(
-            workers = workers,
-            inputs = PipelineConfigInputs(),
-            outputs = PipelineConfigOutputs()
+            workers=workers,
+            inputs=PipelineConfigInputs(),
+            outputs=PipelineConfigOutputs()
         )
         self.config_id = None
         self.config_version = None
-        # self.workers = workers
-        # print("config: " + str(self.pipeline_config))
+        self.attached_connectors = []
+        self.pipeline_id = None
+        self.name = name
+        self.description = description
 
-    def add_input(self, name: str, input_endpoint_config: InputEndpointConfig):
+    def add_input(self, stream: str, input_endpoint_config: DBSPConnector):
         """Add an input endpoint to the pipeline configuration.
 
         Args:
-            name (str): Endpoint name (must be unique across input endpoints).
+            stream (str): Input name (what table to connect to).
```

Comment:
This is related to the other issue in `runner.rs`: we need both stream (i.e., table) name and unique endpoint name. Is the plan to use `uuid` for the latter? If so, then this is good.

Reply:
Yes, I changed it to uuid.